File: ruby/easy/ugly_number.rb
# @param {Integer} num
# @return {Boolean}
def is_ugly(num)
    return false unless num > 0
    factors = [2,3,5]
    while num > 1
        prev = num
        factors.each do |factor|
            num /= factor if (num % factor) == 0
        end
        return false if prev == num
    end
    return true
end

File: javascript/easy/uglyNumber.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isUgly = function(num) {
    if(num <= 0) return false;
    let factors = [2,3,5];
    while(num > 1) {
        let temp = num;
        for(let factor of factors) {
            if((num % factor) == 0) {
                num = parseInt(num/factor);
            }
        }
        if(num === temp) {
            return false;
        }
    }
    return true;
};
